File: Prog 3/Trabalho 1/src/trab_java/Candidato.java
package trab_java;

public class Candidato implements Comparable<Candidato> {
	//Attributes
	private String nome;
	private int votos;
	private Partido partido;
	private boolean eleito;
	
	//Constructor
	public Candidato(String nome,int votos, Partido partido, boolean eleito) {
		this.nome = nome;
		this.votos = votos;
		this.partido = partido;
		this.eleito = eleito;
		
		//O candidato é adicionado ao seu partido no momento de sua criação
		partido.addCandidato(this);
	}
		
	//Getters
	public int getVotos() {
		return this.votos;
	}
	
	public boolean isEleito() {
		return this.eleito;
	}
	
	//Methods
	@Override
	public String toString() {
		String txt = this.nome;
		
		txt += " (" + this.partido.getNome() + ", " + this.votos + " votos)";
		
		if( this.partido.getColigacao().tamanho() > 1) txt += " - Coligação: " + this.partido.getColigacao().getNome();
				
		return txt;
	}

	@Override
	public int compareTo(Candidato x) {
		return x.getVotos() - this.getVotos();
	}
}

File: Programação 3/Trabalho 2/src/Candidato.cpp
#include "Candidato.h"
#include <string>

Candidato::Candidato(string nome, int votos, Partido* p, bool eleito) {
	this->nome = nome;
	this->votos = votos;
	this->eleito = eleito;
	p->addCandidatoData(votos,eleito);
	partido = p;
}

Candidato::Candidato(){
	nome = nullptr;
	votos = 0;
	eleito = false;
	partido = nullptr;
}

Candidato::Candidato(const Candidato& c){
	nome = c.nome;
	votos = c.votos;
	eleito = c.eleito;
	partido = c.partido;
}

const string Candidato::getNome() const {return nome;}

const int Candidato::getVotos() const {return votos;}

const bool Candidato::isEleito() const {return eleito;}

const Partido* Candidato::getPartido() const {return partido;}

bool Candidato::operator<(const Candidato& c) const {
	return getVotos() < c.getVotos() ? true : false;
}

Candidato::~Candidato() {}
